editor/filesystem_dock.cpp

```diff
@@ -537,7 +537,7 @@ void FileSystemDock::_notification(int p_what) {
 
 			_update_display_mode();
 
-			if (EditorFileSystem::get_singleton()->is_scanning()) {
+			if (EditorFileSystem::get_singleton()->is_scanning() || EditorFileSystem::get_singleton()->doing_first_scan()) {
```

Comment:
Doesn't `doing_first_scan()` imply `is_scanning()`?

Reply:
I was not sure if `EditorFileSystem::get_singleton()->is_scanning()` was still necessary. I guess not, I don't think it's possible to get a ready notification while scanning but not being the first scan.
Done!

Reply:
I meant it other way around, i.e. if `doing_first_scan()` is true, it means a scan is in progress, so `is_scanning()` should also be true. Checking both is redundant.

Reply:
In fact, before, it was only `is_scanning()` but it caused the problem of not displaying le progress bar because on ready of the FileSystemDock, `is_scanning` returns false. `first_scan` is set to true on the creation of `EditorFileSystem` while `scanning` is set to true only when `scan` is call which is on a deferred call from `EditorNode`.
Maybe `scanning` should be true by default??

Reply:
Once again I missed the point of this condition .-.
`is_scanning()` could have an added check for `first_scan`, but not sure if it won't break things. You can go back to the double check like before, as it was working correctly.

Reply:
I see your point, effectively, `is_scanning` should returns true when `first_scan` is true. I'm just worry about `ResourceImporterTexture::update_imports()` that checks `is_scanning`. Could adding `first_scan` in `is_scanning` have unwanted side effects?

Reply:
I tested it and I think your suggestion makes a lot of sense. It prevents updating texture imports while first scan is running. I changed `is_scanning` to include `first_scan` and used `is_scanning` in `FileSystemDock`.